### PR TITLE
Mw 87/create progress bar component

### PR DESCRIPTION
**Pull request by Alex89198900**
1.Create progress bar component.
2. Link to render.
3.Add style

### CODE REVIEW

File: mw-migrations/.env.example
Comment (tritonJS826) [inline mw-migrations/.env.example:1]: why do you removed .env.exmample from migrations?

File: mw-migrations/backups/README.md
Comment (tritonJS826) [inline mw-migrations/backups/README.md:1]: please restore this file

File: mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_clearAllCollections.log.txt
Comment (tritonJS826) [inline mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_clearAllCollections.log.txt:1]: could you remove migration log please, I thinkit is not neccessary for now.

File: mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_firebase.log.txt
Comment (tritonJS826) [inline mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_firebase.log.txt:1]: could you remove migration log please, I thinkit is not neccessary for now.

File: mw-migrations/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "migrate-dayReport-deprecated": "ts-node ./src/migrations/dayReportsMigration-delete-deprecated-fields.ts",
     "migrate-way-formerMentorUuids": "ts-node ./src/migrations/waysMigration-add-formerMentorUuids.ts",
     "export-backup": "ts-node ./src/backup/export.ts",
-    "import-backup":  "ts-node ./src/backup/import.ts" 
+    "import-backup":  "node --loader ts-node/esm ./src/backup/import.ts"
```
Comment (tritonJS826) [inline mw-migrations/package.json:20]: whar does this changes do?

File: mw-migrations/pnpm-lock.yaml
Comment (tritonJS826) [inline mw-migrations/pnpm-lock.yaml:1]: why did you remove it?

File: mw-webapp/.env.example
Comment (tritonJS826) [inline mw-webapp/.env.example:1]: why did you remove it?

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * ProgressBar component
+ */
+export const ProgressBar = (arg: ProgressBarProps) => {
+  const PERCENT = 100;
+  const countTrue = arg.progress.filter((item) => item).length;
+  const computedProcent = (PERCENT / arg.progress.length) * countTrue;
+  if (arg.progress.length > 0) {
+    return (
+      <>
```
Comment (tritonJS826) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:16]: unneccessary react fragment

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import styles from "src/component/progressBar/ProgressBar.module.scss";
+ type ProgressBarProps = {/**
+                           * ProgressBar prop type
+                           */
+   progress: Array<boolean>;}
```
Comment (tritonJS826) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:5]: looks like broken formatting please check

Comment (tritonJS826) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:5]: From the external perspective it is not clear what the boolean list required for render progress bar

Comment (tritonJS826) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:5]: what if it is empty? This behavior not specified

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * ProgressBar component
+ */
+export const ProgressBar = (arg: ProgressBarProps) => {
+  const PERCENT = 100;
```
Comment (tritonJS826) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:11]:
```suggestion
  const MAX_PERCENTAGE = 100;
```

**Review by tritonJS826 — COMMENTED** [review]
Ok, pretty good. But let's think about the components like about reusable small components which you can use everywhere (kind of library). Think more deeply about external interface. Lets implement this external interface 
```
type progressBarProps = {

/**
* Text to show in progress bar
* No text by default
*/
text?: string

/**
* Fill percentage 
* (minimum 0 maximum 100)
*/
percentage: number;
}
```

For better understanding how to create components check different libraries (just for example: https://react-bootstrap.netlify.app/docs/components/progress/, https://ant.design/components/progress,). Check interfaces and functionality in first place

File: mw-webapp/src/component/progressBar/ProgressBar.module.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
+@import "src/styles/variables";
+
+$progressBarContainerHeigh: 20px;
+
+.progressBarContainer {
+  height: $progressBarContainerHeigh;
+  border-radius: 5px;
+  margin: 5px 0px 5px 0px;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/progressBar/ProgressBar.module.scss:8]: Please use global vars. Check everywhere)

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import styles from "src/component/progressBar/ProgressBar.module.scss";
+ type ProgressBarProps = {/**
+                           * ProgressBar prop type
+                           */
+   progress: Array<boolean>;}
+
+/**
+ * ProgressBar component
+ */
+export const ProgressBar = (arg: ProgressBarProps) => {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:10]: arg => props

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import styles from "src/component/progressBar/ProgressBar.module.scss";
+ type ProgressBarProps = {/**
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:2]: type => interface

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 4 above)
Comment (Ekaterina1994) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:11]: Move const before component

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/progressBar/ProgressBar.module.scss [hunk 8]
```diff
@@ -0,0 +1,22 @@
+@import "src/styles/variables";
+
+$progressBarContainerHeigh: 20px;
+
+.progressBarContainer {
+  height: $progressBarContainerHeigh;
```
Comment (SergioAJS) [inline mw-webapp/src/component/progressBar/ProgressBar.module.scss:6]: missing 't' letter in the  word Height (progressBarContainerHeigh**t**)

Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.module.scss:6]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 4 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 7 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx:2]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 6 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx:11]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.module.scss (hunk 5 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.module.scss:8]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 4 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 3 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx:5]: Done

Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx:5]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 2 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx]: Done

File: mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_firebase.log.txt
Comment (Alex89198900) [reply mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_firebase.log.txt:1]: Done

File: mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_clearAllCollections.log.txt
Comment (Alex89198900) [reply mw-migrations/migration-logs/2024-01-17T18-46-27.102Z_clearAllCollections.log.txt:1]: Done

File: mw-migrations/backups/README.md
Comment (Alex89198900) [reply mw-migrations/backups/README.md:1]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.module.scss [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+$progressBarContainerHeight: 20px;
+$progressBarBorderRadius: 20px;
+
+.progressBarContainer {
+  height: $progressBarContainerHeight;
+  border-radius: $progressBarBorderRadius;
+  margin-top: $marginSmall;
+  margin-bottom: $marginSmall;
+  background-color: rgb(74, 74, 70);
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/progressBar/ProgressBar.module.scss:11]: For colors we use vars from Theme

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+   * (minimum 0 maximum 100)
+   */
+  percentage: number;
+}
+
+/**
+ * ProgressBar component
+ */
+export const ProgressBar = (props: ProgressBarProps) => {
+  if (!isNaN(props.percentage)) {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:25]: Do we need this check?  Is it possible that props.percentage is not number?  IN interface percentage is only number

File: mw-webapp/src/component/progressBar/ProgressBar.module.scss (hunk 9 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.module.scss]: Done

File: mw-webapp/src/component/progressBar/ProgressBar.tsx (hunk 10 above)
Comment (Alex89198900) [reply mw-webapp/src/component/progressBar/ProgressBar.tsx]: Done

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-migrations/src/backup/README.md
Comment (tritonJS826) [inline mw-migrations/src/backup/README.md:1]: why moved?

File: mw-webapp/src/component/progressBar/ProgressBar.tsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    <div className={styles.progressBarContainer}>
+      <div
+        style={{width: `${props.percentage}%`}}
+        className={styles.progressIndicator}
+      >
+        <div className={styles.progressIndicatorPercent}>
+          {props.text}
+        </div>
+      </div>
+
```
Comment (tritonJS826) [inline mw-webapp/src/component/progressBar/ProgressBar.tsx:35]: unneccessary new line

File: mw-webapp/src/logic/wayPage/GoalMetricsBlock.tsx [hunk 12]
```diff
@@ -81,8 +84,15 @@ export const GoalMetricsBlock = (props: GoalMetricStatisticsBlockProps) => {
    * Render goal metrics
    */
   const renderGoalMetrics = (goalMetric: GoalMetric) => {
+    const countTrue = goalMetric.isDone.filter((item) => item).length;
+    const computedProcent = countTrue > 0 ? (MAX_PERCENTAGE / goalMetric.isDone.length) * countTrue : 0;
+
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/GoalMetricsBlock.tsx:89]:
```suggestion
    const doneMetricsAmount = goalMetric.isDone.filter(Boolean).length;
    const computedProcent = doneMetricsAmount && Math.round(countTrue / goalMetric.metricUuids.length * MAX_PERCENTAGE);

```

Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/GoalMetricsBlock.tsx:89]: Done

File: mw-migrations/src/backup/README.md
Comment (Alex89198900) [reply mw-migrations/src/backup/README.md:1]: Done

**Review by tritonJS826 — APPROVED** [review]
